pwa/app/components/tba/searchbar.tsx

```diff
@@ -0,0 +1,135 @@
+import { useNavigate } from '@remix-run/react';
+import { useEffect, useMemo, useRef, useState } from 'react';
+
+import { SearchIndex } from '~/api/v3';
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+} from '~/components/ui/command';
+import FuzzysortFilterer from '~/lib/search/fuzzysortFilterer';
+import { ProdAPIProvider } from '~/lib/search/prodAPIProvider';
+import { cn } from '~/lib/utils';
+
+export default function Searchbar() {
+  const navigate = useNavigate();
+  const [query, setQuery] = useState('');
+  const [isOpen, setIsOpen] = useState(false);
+
+  const provider = useMemo(() => new ProdAPIProvider(), []);
+  const filterer = useMemo(() => new FuzzysortFilterer(), []);
+
+  const [fullSearchData, setFullSearchData] = useState<SearchIndex | null>(
+    null,
+  );
+
+  useEffect(() => {
+    provider
+      .provide()
+      .then((data) => {
+        setFullSearchData(data);
+      })
+      .catch(() => {
+        // todo: log in sentry
+      });
+  }, [provider]);
+
+  const [searchResults, setSearchResults] = useState<SearchIndex | null>(null);
+
+  useEffect(() => {
+    if (fullSearchData === null) {
+      return;
+    }
+
+    const searchResults = filterer.filter(fullSearchData, query);
+    setSearchResults(searchResults);
+  }, [query, fullSearchData, filterer]);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  // Toggle the menu when ⌘K or ctrl-k is pressed
+  useEffect(() => {
+    function listener(e: KeyboardEvent) {
+      if (e.key === 'k' && (e.ctrlKey || e.metaKey)) {
+        e.preventDefault();
+        setIsOpen((o) => !o);
+        inputRef.current?.focus();
```

Comment:
It should be `focus` on open but `blur` on close.

Reply:
Maybe consider use `<Popover />`, it'll handle you all the focus things like auto focus, blur, and focus trapping.